### ISSUE TITLE
Support for ASP.NET Core RC2

### ISSUE THREAD

**Issue by IEvangelist** [domaindrivendev/Ahoy#84]
Currently this only supports up to **ASP.NET Core RC1**. We were using this for our **RC1** source, but since we have updated to **ASP.NET Core RC2** -- this is no longer compatible at all.  :cry:

**Comment by domaindrivendev** [comment]
Looking at the various PR's, it looks like a lot of people are duplicating work - I also have a local RC2 branch in progress. My bad for lack of communication. With two small kids there just aint enough hours in the day - even to just get the appropriate communication out there :(

I won't be accepting any PR's that don't come with a green suite of tests so that makes #88 the likely candidate. I'm going to pull it down tomorrow and compare with my own local branch before merging one of them into master. Either way, I'm targeting 5/28 for an RC2 compatible release to Nuget. See the beta2 milestone here https://github.com/domaindrivendev/Ahoy/milestones. Note the revert to "beta" status - still lots of feature gaps and a config. API that will likely undergo change before we reach a stable release.

Thanks to everyone for chipping in. Please don't get disheartened if I'm slow to respond to PR's. So long as they include tests and a passing suite, I'll get always consider them and strive to respond as soon as I can.

I have one significant architectural refactor that I want to get done after the beta2 release. After this, I'll be looking to open up the project and add additional contributors. For anyone reading this, please let me know if you're interested?

Ahoy!

Richie

**Comment by domaindrivendev** [comment]
eeny meeny miny moe - let's pick this one as the canonical issue for RC2 support

**Comment by Zoltu** [comment]
Is there a status update on this?  I read in another issue a target NuGet package release of 5/28.  My project is currently blocked on this (I need some things out of RC2 but I can't upgrade until this is RC2 compatible).  If this is still a ways out I may try to get one of the many forks out there compiling and building into a NuGet package in my private feed... but I would much rather use an official release even if it isn't fully ready yet.

**Comment by domaindrivendev** [comment]
Gonna be a day late - as in will be released today. I'm there short of literally just pressing the button. But, will be away from my laptop for next several hours so I'd say check back later or tomorrow morning at it should be available.

**Comment by ronnyek** [comment]
I was able to install beta9 (I've never had swagger setup with anything) and now I'm getting stuff populated in my swagger.json. It was complaining about xml path stuff that i fixed, and made sure I'm spitting out xml at the path its looking. That's all I manage to get out of swagger no matter how many different changes I've made to my controllers. Is there some minimum config that has to be there in order for reflection / xml stuff to work? Inherit from special controller, or ApiController? Maybe an attribute I'm missing?

{
"swagger": "2.0",
"info": {
"version": "v1",
"title": "Core Identity Management API",
"description": "A sample API for testing Swashbuckle",
"termsOfService": "Some terms ..."
},
"basePath": "/",
"paths": {},
"definitions": {},
"securityDefinitions": {}
}

![image](https://cloud.githubusercontent.com/assets/584713/15660572/355488ba-2693-11e6-85e5-a1b5bd3c0393.png)

**Comment by Zoltu** [comment]
This issue can probably be closed.  I was able to get everything working in RC2 with `-beta9` NuGet package.  I did have to tweak my configuration, looks like the structure changed a bit.

New Configuration:
```
services.AddSwaggerGen(options =>
{
	options.SingleApiVersion(new Info { Version = "v1", Title = "My Title" });
	options.DescribeAllEnumsAsStrings();
});
```

@ronnyek The above is in `Startup.cs` `ConfigureServices` method and should be all the configuration you need to get things working.

**Comment by ronnyek** [comment]
I have all that, but alas, the json posted above is all I get out of it. I have controllers that dont inherit from anything, but tried inheriting from controller, tried the produces attributes, tried the explicit [SwaggerResponse], [SwaggerOperation]. Also, originally these were in a controller in an area, but I pulled it out and put it into its local controller. Still get nothing but the info in that json. Not seeing any errors either =(

**Comment by Zoltu** [comment]
@ronnyek My controllers look like this and they are picked up auto-magically:
```
[Route("api/my")]
public class MyController : Controller
{
	[HttpGet]
	[Route("")]
	public async Task<IActionResult> Get()
	{
		return Ok("Yay!");
	}
}
```

**Comment by ronnyek** [comment]
Ahh the trick was the implicit route in there... looks good to me too!

**Comment by bragma** [comment]
Everything's working file with 6.0.0-beta9! Thanks a lot!
Regarding getting the xml file path, I am using this on framework new461:

```
private string GetXmlCommentPath()
{
	var app = PlatformServices.Default.Application;
	return Path.Combine(app.ApplicationBasePath, Path.ChangeExtension(app.ApplicationName, "xml"));
}
```
